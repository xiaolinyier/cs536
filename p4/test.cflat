struct point {
	int x;
	int y;
};

int manhattanDist(int ax, int ay, int bx, int by) {
	return ax - bx + ay - by;
}

int main() {
	struct point a;
	struct point b;

	cin >> a.x;
	cin >> a.y;
	cin >> b.x;
	cin >> b.y;
}

// in class example

struct Point {
    int x;
    int y;
};

int f(int x, bool b) { }
void g() {
    int a;
    bool b;
    struct Point p;
    p.x = a;
    b = a == 3;
    f(a + p.y*2, b);
    g();
}

//valid struct
struct st {
	int validVar;
};

// valid function 
void undeclaredNames() {
	// valid names
	int val0;
	int val1;
	bool val2;
	bool val3;
	struct st inst0;
	
	bool abc;
	
	// if statement
	if(abc) { }
	if (inst0.validVar) { }
	
	// if else statement
	if(abc) { } else { }
	if (inst0.validVar) { } else { }
	
	// while statement
	while(abc) { }
	while (inst0.validVar) { }
	
	// repeat statement
	repeat (abc) { }
	repeat (inst0.validVar) { }

	// operations then assignment of unkown names
	val0 = val0 + val1;
	val0 = val1 + val0;
	val0 = val0 - val1;
	val0 = val1 - val0;
	val0 = val0 * val1;
	val0 = val1 * val0;
	val0 = val0 / val1;
	val0 = val1 / val0;
	
	val2 = val0 < val1;
	val2 = val1 < val0;
	val2 = val0 > val1;
	val2 = val1 > val0;
	val2 = val0 <= val1;
	val2 = val1 <= val0;
	val2 = val0 >= val1;
	val2 = val1 >= val0;
	val2 = val0 == val1;
	val2 = val1 == val0;
	val2 = val0 != val1;
	val2 = val1 != val0;
	
	// unary and write operators
	val1++;
	val1--;
	cout << val1;
	cin >> val1;

	// bools
	val2 = val2 && val3;
	val2 = val3 && val2;
	val2 = val2 || val3;
	val2 = val3 || val2;
	val2 = !val3;
	
	// test in funciton call
	undeclaredNames(val1);
	undeclaredNames(inst0.validVar);
}