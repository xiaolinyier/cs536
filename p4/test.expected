struct point{
    int x;
    int y;
};

int manhattanDist(int ax, int ay, int bx, int by) {
    return (((ax(int) - bx(int)) + ay(int)) - by(int));
}

int main() {
    struct point a;
    struct point b;
    cin >> a(point).x(int);
    cin >> a(point).y(int);
    cin >> b(point).x(int);
    cin >> b(point).y(int);
}

struct Point{
    int x;
    int y;
};

int f(int x, bool b) {
}

void g() {
    int a;
    bool b;
    struct Point p;
    p(Point).x(int) = a(int);
    b(bool) = (a(int) == 3);
    f(int,bool->int)((a(int) + (p(Point).y(int) * 2)), b(bool));
    g(->void)();
}

struct st{
    int validVar;
};

void undeclaredNames() {
    int val0;
    int val1;
    bool val2;
    bool val3;
    struct st inst0;
    bool abc;
    if (abc(bool)) {
    }
    if (inst0(st).validVar(int)) {
    }
    if (abc(bool)) {
    }
    else {
    }
    if (inst0(st).validVar(int)) {
    }
    else {
    }
    while (abc(bool)) {
    }
    while (inst0(st).validVar(int)) {
    }
    repeat (abc(bool)) {
    }
    repeat (inst0(st).validVar(int)) {
    }
    val0(int) = (val0(int) + val1(int));
    val0(int) = (val1(int) + val0(int));
    val0(int) = (val0(int) - val1(int));
    val0(int) = (val1(int) - val0(int));
    val0(int) = (val0(int) * val1(int));
    val0(int) = (val1(int) * val0(int));
    val0(int) = (val0(int) / val1(int));
    val0(int) = (val1(int) / val0(int));
    val2(bool) = (val0(int) < val1(int));
    val2(bool) = (val1(int) < val0(int));
    val2(bool) = (val0(int) > val1(int));
    val2(bool) = (val1(int) > val0(int));
    val2(bool) = (val0(int) <= val1(int));
    val2(bool) = (val1(int) <= val0(int));
    val2(bool) = (val0(int) >= val1(int));
    val2(bool) = (val1(int) >= val0(int));
    val2(bool) = (val0(int) == val1(int));
    val2(bool) = (val1(int) == val0(int));
    val2(bool) = (val0(int) != val1(int));
    val2(bool) = (val1(int) != val0(int));
    val1(int)++;
    val1(int)--;
    cout << val1(int);
    cin >> val1(int);
    val2(bool) = (val2(bool) && val3(bool));
    val2(bool) = (val3(bool) && val2(bool));
    val2(bool) = (val2(bool) || val3(bool));
    val2(bool) = (val3(bool) || val2(bool));
    val2(bool) = (!val3(bool));
    undeclaredNames(->void)(val1(int));
    undeclaredNames(->void)(inst0(st).validVar(int));
}

